modules/civiremote_event/src/Element/Confirm.php

```diff
@@ -0,0 +1,118 @@
+<?php
+/*
+ * Copyright (C) 2023 SYSTOPIA GmbH
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as published by
+ *  the Free Software Foundation in version 3.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+declare(strict_types=1);
+
+namespace Drupal\civiremote_event\Element;
+
+use Drupal\Core\Render\Element\FormElement;
+use Drupal\Core\Form\FormStateInterface;
+
+/**
+ * Provides a form element for double-input of values.
+ *
+ * Formats as a pair of fields, which do not validate unless the two entered
+ * values match.
+ *
+ * Properties:
+ * - #field: The render array of the form field to confirm.
+ * - #confirm_title: The title of the confirmation field (optional).
+ *
+ * @FormElement("confirm")
+ */
+final class Confirm extends FormElement {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getInfo(): array {
+    return [
+      '#field' => [],
+      '#confirm_title' => NULL,
+      '#input' => TRUE,
+      '#markup' => '',
+      '#process' => [
+        [static::class, 'processConfirm'],
+      ],
+      '#theme_wrappers' => [
+        'form_element',
+      ],
+    ];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function valueCallback(&$element, $input, FormStateInterface $form_state): array {
+    if ($input === FALSE) {
+      return $element['#default_value'] = [
+        'value1' => $element['#field']['#default_value'] ?? NULL,
+        'value2' => $element['#field']['#default_value'] ?? NULL,
+      ];
+    }
+
+    /** @phpstan-var array<string, mixed> $input */
+    return [
+      'value1' => $input['value1'] ?? $element['#field']['#default_value'] ?? NULL,
+      'value2' => $input['value2'] ?? $element['#field']['#default_value'] ?? NULL,
+    ];
```

Comment:
I think it would make sense to call the `valueCallback` of the element type's render element, as there might be transformations necessary for specific field types.

Reply:
Maybe we should run `getValueCallback()` in the process callback of the element, so that there's validation for whether the element given really is a form element. So let's add the value callback as a `#` property in `processConfirm()` and just call it from there - what do you think?